presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -939,35 +1009,54 @@ private InputStream openStream(Path path, long start)
                             }
                         });
             }
-            catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-                throw new RuntimeException(e);
-            }
             catch (Exception e) {
-                throwIfInstanceOf(e, IOException.class);
-                throwIfUnchecked(e);
-                throw new RuntimeException(e);
+                throw propagate(e);
             }
         }
 
         private void closeStream()
         {
             if (in != null) {
-                try {
-                    if (in instanceof S3ObjectInputStream) {
-                        ((S3ObjectInputStream) in).abort();
-                    }
-                    else {
-                        in.close();
-                    }
-                }
-                catch (IOException | AbortedException ignored) {
-                    // thrown if the current thread is in the interrupted state
-                }
+                abortStream(in);
                 in = null;
                 STATS.connectionReleased();
             }
         }
+
+        private void checkClosed()
+                throws IOException
+        {
+            if (closed.get()) {
+                throw new EOFException(FSExceptionMessages.NEGATIVE_SEEK);
```

Comment:
Maybe there is more meaningful exception code (e.g: stream closed etc)

Reply:
Thanks, I meant to use `STREAM_IS_CLOSED` here.